tests/pipelinesrun_test.go

```diff
@@ -0,0 +1,209 @@
+package tests
+
+import (
+	"fmt"
+	"github.com/jfrog/jfrog-client-go/pipelines/services"
+	"github.com/jfrog/jfrog-client-go/utils/io/httputils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"github.com/stretchr/testify/assert"
+	"net/http"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestPipelinesRunService(t *testing.T) {
+	initPipelinesTest(t)
+	t.Run("test trigger pipeline resource sync", testTriggerSync)
+	t.Run("test get sync status", testGetSyncStatus)
+	t.Run("test get run status", testGetRunStatus)
+}
+
+const (
+	// define default wait time
+	defaultMaxWaitMinutes    = 10 * time.Minute
+	defaultSyncSleepInterval = 5 * time.Second // 5 seconds
+)
+
+func testTriggerSync(t *testing.T) {
+	if *PipelinesVcsToken == "" {
+		assert.NotEmpty(t, *PipelinesVcsToken, "cannot run pipelines tests without vcs token configured")
+		return
+	}
+	// Create integration with provided token.
+	unixTime := time.Now().Unix()
+	timeString := strconv.Itoa(int(unixTime))
+	integrationName := strings.Join([]string{"github", "integration_test", timeString}, "_")
+	integrationId, err := testsPipelinesIntegrationsService.CreateGithubIntegration(integrationName, *PipelinesVcsToken)
+	assert.NoError(t, err)
+	defer deleteIntegrationAndAssert(t, integrationId)
+
+	// Create source with the above integration and assert.
+	sourceId, srcErr := testsPipelinesSourcesService.AddSource(integrationId, *PipelinesVcsRepoFullPath, *PipelinesVcsBranch, services.DefaultPipelinesFileFilter)
+	assert.NoError(t, srcErr)
+	defer deleteSourceAndAssert(t, sourceId)
+	pollSyncPipelineSource(t)
+}
+
+func testGetSyncStatus(t *testing.T) {
+	if *PipelinesVcsToken == "" {
+		assert.NotEmpty(t, *PipelinesVcsToken, "cannot run pipelines tests without vcs token configured")
+		return
+	}
+	// Create integration with provided token.'
+	unixTime := time.Now().Unix()
+	timeString := strconv.Itoa(int(unixTime))
+	integrationName := strings.Join([]string{"github", "integration_test", timeString}, "_")
+	integrationId, err := testsPipelinesIntegrationsService.CreateGithubIntegration(integrationName, *PipelinesVcsToken)
+	assert.NoError(t, err)
+	defer deleteIntegrationAndAssert(t, integrationId)
+
+	// Create source with the above integration and assert.
+	sourceId, srcErr := testsPipelinesSourcesService.AddSource(integrationId, *PipelinesVcsRepoFullPath, *PipelinesVcsBranch, services.DefaultPipelinesFileFilter)
+	assert.NoError(t, srcErr)
+
+	defer deleteSourceAndAssert(t, sourceId)
+	pollSyncPipelineSource(t)
+	pollForSyncResourceStatus(t)
+}
+
+func testGetRunStatus(t *testing.T) {
+	assert.NotEmpty(t, *PipelinesVcsToken, "cannot run pipelines tests without vcs token configured")
```

Comment:
To be consistent with the other tests
```suggestion
	if !assert.NotEmpty(t, *PipelinesVcsToken, "cannot run pipelines tests without vcs token configured") {
		return
	}
```